src/libbson/tests/test-json.c

```diff
@@ -2841,6 +2841,538 @@ test_bson_as_json_multi_object (void)
+
+   bson_destroy (b);
+   bson_destroy (scope);
+}
+
+static void
+test_bson_as_json_with_opts_int32 (void)
+{
+   bson_t *b;
+
+   b = bson_new ();
+   BSON_ASSERT (BSON_APPEND_INT32 (b, "v", 461394000));
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_RELAXED,
+			   "{ \"v\" : 461394000 }");
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_CANONICAL,
+			   "{ \"v\" : { \"$numberInt\" : \"461394000\" } }");
+
+   bson_destroy (b);
+}
+
+static void
+test_bson_as_json_with_opts_int64 (void)
+{
+   bson_t *b;
+
+   b = bson_new ();
+   BSON_ASSERT (BSON_APPEND_INT64 (b, "v", 461394000));
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_RELAXED,
+			   "{ \"v\" : 461394000 }");
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_CANONICAL,
+			   "{ \"v\" : { \"$numberLong\" : \"461394000\" } }");
+
+   bson_destroy (b);
+}
+
+static void
+test_bson_as_json_with_opts_timestamp (void)
+{
+   bson_t *b;
+
+   b = bson_new ();
+   BSON_ASSERT (BSON_APPEND_TIMESTAMP (b, "v", 461394000, 2));
+
+   run_bson_as_json_with_opts_tests (
+      b,
+      BSON_JSON_MODE_RELAXED,
+      "{ \"v\" : { \"$timestamp\" : { \"t\" : 461394000, \"i\" : 2 } } }");
+
+   run_bson_as_json_with_opts_tests (
+      b,
+      BSON_JSON_MODE_CANONICAL,
+      "{ \"v\" : { \"$timestamp\" : { \"t\" : 461394000, \"i\" : 2 } } }");
+
+   bson_destroy (b);
+}
+
+static void
+test_bson_as_json_with_opts_minkey (void)
+{
+   bson_t *b;
+
+   b = bson_new ();
+   BSON_ASSERT (BSON_APPEND_MINKEY (b, "v"));
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_RELAXED,
+			   "{ \"v\" : { \"$minKey\" : 1 } }");
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_CANONICAL,
+			   "{ \"v\" : { \"$minKey\" : 1 } }");
+
+   bson_destroy (b);
+}
+
+static void
+test_bson_as_json_with_opts_maxkey (void)
+{
+   bson_t *b;
+
+   b = bson_new ();
+   BSON_ASSERT (BSON_APPEND_MAXKEY (b, "v"));
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_RELAXED,
+			   "{ \"v\" : { \"$maxKey\" : 1 } }");
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_CANONICAL,
+			   "{ \"v\" : { \"$maxKey\" : 1 } }");
+
+   bson_destroy (b);
+}
+
+static void
+test_bson_as_json_with_opts_decimal128 (void)
+{
+   bson_t *b;
+   bson_decimal128_t dec;
+
+   dec.high = 0x3040ffffffffffffULL;
+   dec.low = 0xffffffffffffffffULL;
+
+   b = bson_new ();
+   BSON_ASSERT (BSON_APPEND_DECIMAL128 (b, "v", &dec));
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_RELAXED,
+			   "{ \"v\" : { \"$numberDecimal\" : "
+			   "\"5192296858534827628530496329220095\" } }");
+
+   run_bson_as_json_with_opts_tests (b,
+			   BSON_JSON_MODE_CANONICAL,
+			   "{ \"v\" : { \"$numberDecimal\" : "
+			   "\"5192296858534827628530496329220095\" } }");
+
+   bson_destroy (b);
+}
+
+static void
+test_bson_as_json_with_opts_all_types (void)
+{
+   const char *full_canonical;
+   const char *full_relaxed;
```

Comment:
`full_canonical` and `full_relaxed` should be non-const `char *`. Currently `debug-compile-with-warnings` task fails due to a warning - L3369 is freeing a `const char*`.

Reply:
Done.